pty/src/win/psuedocon.rs

```diff
@@ -107,6 +107,70 @@ impl PsuedoCon {
         Ok(())
     }
 
+    pub fn spawn_command_as_user(&self, cmd: CommandBuilder, h_token: HANDLE) -> anyhow::Result<WinChild>  {
```

Comment:
I'm not keen on the amount of code duplication between this and the existing `spawn_command`.

Can you refactor `spawn_command` to accept an `Option<HANDLE>` for the token?

I also noticed that the environment block and cwd aren't passed down in this case; please either enable that or add some comments explaining why that behavior is different.

Reply:
Hi @wez, thanks for taking a look at my PR! I've gone ahead and made the changes you've suggested. Let me know if there is anything else you need. Thanks!